lib/utils/Utils.js

```diff
@@ -205,6 +205,24 @@ export default class Utils {
   }
 
   /**
+   * Appends a query parameter to an existing URL, taking care of existing query parameters.<br />
+   * This method returns `uri` as-is if `key` or `value` is not defined.
+   *
+   * @param uri {String} The URI to modify.
+   * @param key {String} The name of the parameter to append.
+   * @param value {String} The value of the parameter to append.
+   *
+   * @returns {String} The modified URI.
+   */
+  static appendQueryParameter(uri, key, value) {
+    if (!uri || !key || !value) {
+      return uri;
+    }
+
+    return uri + (uri.indexOf('?') > -1 ? '&' : '?') + encodeURIComponent(key) + '=' + encodeURIComponent(value);
```

Comment:
I think that this function should be really enhanced to deal with different issues: spaces and special characters. WDYT?